components/apimgt/org.wso2.carbon.apimgt.rest.api.publisher/src/main/java/org/wso2/carbon/apimgt/rest/api/publisher/utils/ApiImportExportManager.java

```diff
@@ -70,24 +70,35 @@ public ApiImportExportManager(APIPublisher apiPublisher) {
         // iterate and collect all information
         for (API api : apis) {
             api = apiPublisher.getAPIbyUUID(api.getId());
-            // get endpoints
-            Map<String, String> endpoints = api.getEndpoint();
+            // get endpoints at API Level
+            Map<String, Object> endpoints = api.getEndpoint();
             if (endpoints.isEmpty()) {
                 log.error("No Endpoints found for api: " + api.getName() + ", version: " + api.getVersion());
                 // skip this API
                 continue;
             }
             Set<Endpoint> endpointSet = new HashSet<>();
-            try {
-                for (Map.Entry<String, String> endpointEntry : endpoints.entrySet()) {
-                    endpointSet.add(apiPublisher.getEndpoint(endpointEntry.getValue()));
+            for (Map.Entry<String, Object> endpointEntry : endpoints.entrySet()) {
+                if (endpointEntry.getValue() instanceof Endpoint) {
+                    endpointSet.add((Endpoint) endpointEntry.getValue());
+                } else {
+                    endpointSet.add(apiPublisher.getEndpoint(String.valueOf(endpointEntry.getValue())));
                 }
-            } catch (APIManagementException e) {
-                log.error("Error in getting endpoints for api: " + api.getName() + ", version: " + api.getVersion(), e);
-                // skip this API
-                continue;
             }
-
+            // get Endpoints at Resource Level
+            api.getUriTemplates().forEach((k, v) -> {
+                v.getEndpoint().forEach((type, value) -> {
+                    if (value instanceof Endpoint) {
+                        endpointSet.add((Endpoint) value);
+                    } else {
+                        try {
+                            endpointSet.add(apiPublisher.getEndpoint(String.valueOf(value)));
+                        } catch (APIManagementException e) {
+                            log.error("Error in getting endpoints for Resource: " + v.getTemplateId(), e);
```

Comment:
here error is not thrown. If this was intentionally done, can you put a comment explaining the reason